### DISCUSSION TITLE
CPU - Electrical consumption

### DISCUSSION THREAD

**Discussion by consithe1** [Boavizta/boaviztapi#200] [Q&A] [answered]
Hello everyone!

I was going through the documentation and it seems like I can't get the power consumption and electrical consumption right from the example given [here](https://github.com/Boavizta/boaviztapi/blob/main/docs/docs/Explanations/usage/elec_conso.md#example-for-a-cpu).

As I understand it, the values taken in this example to calculate the `consumption_profile(workload)` are for the [Intel Xeon E](https://github.com/Boavizta/boaviztapi/blob/main/boaviztapi/data/consumption_profile/cpu/cpu_profile.csv). 

- The first question I have is what are `a`, `b`, `c` and `d`? What do they represent and how to "calculate" them if they are not known?

Based on the documentation, `consumption_profile(workload) = a * ln(b * (workload + c)) + d`
By applying this formula, I get the following results:

| load_percentage | High (100%) | Medium (50%) | Low (10%) | idle (0%) | off |
| --- | --- | --- | --- | --- | --- |
| `consumption_profile(workload)` | 100 W | 70 W | 24 W | 2 W| 0 W |
| theoretical results(found in the [example](https://github.com/Boavizta/boaviztapi/blob/main/docs/docs/Explanations/usage/elec_conso.md#example-for-a-cpu)) | 260 W | 182 W | 77 W | 36 W | 0 W |

- Do you know if I made a mistake and did not apply correctly the formula or if something is missing and was forgotten in the documentation? 

Thank you for your time,
Best regards!

**Comment by da-ekchajzer** [comment, marked as answer]
> The first question I have is what are a, b, c and d? What do they represent and how to "calculate" them if they are not known?

a, b, c are the parameters of a logarithmic function (consumption_profile) which represent the electrical consumption of a CPU depending on his load.

The parameters have been computed through regression on [Teads dataset](https://docs.google.com/spreadsheets/d/1DqYgQnEDLQVQm5acMAhLgHLD8xXCG9BIrk-_Nv6jF3k/) for : 

* Several model ranges
* Based on the TDP

The details can be found here : https://doc.api.boavizta.org/Explanations/components/cpu/#consumption-profile

>  Do you know if I made a mistake and did not apply correctly the formula or if something is missing and was forgotten in the documentation?

There is a mistake on the documentation. What you calculated is right and correspond to what the API will compute. 
It has been corrected here : https://github.com/Boavizta/boaviztapi/commit/6343c9fcd303e9c92a88888d0e3474ae578e6033

While testing, I came up with a fuzzy matching error that also have been corrected : https://github.com/Boavizta/boaviztapi/commit/4a9728fb7125ac45dbe2ade92abb64f3fb8bbab4

Thank you very much for your feedback !

**Reply by da-ekchajzer** [reply]
Don't forget that this is only for CPU. You also need to take into account RAM and other consumptions on the server.

**Reply by consithe1** [reply]
Thank you very much for your reply!
